README.md

```diff
@@ -1072,6 +1072,306 @@ We recommend that you can write your own `CMakeUserPresets.json` file. You can i
 }
 ```
 
+
+## Cheat sheet: autotools vs. cmake
+
+Assume that `$oomph_home_dir` is the `oomph-lib` home directory:
+<table>
+<tr><th>autotools</th><th>cmake</th></tr>
+
+<tr>
+<td colspan="2">
+<b>- Build and install the entire library using a helper script:</b>
+</td>
+</tr>
+<tr>
+<td>
+<pre>
+# Go to home directory
+cd $oomph_home_dir<br>
+# Run the autogen script which prompts
+# for configure options
+./autogen.sh
+
+</pre>
+</td>
+<td>
+<pre>
+# Go to home directory
+cd $oomph_home_dir<br>
+# Run the hierher script which prompts
+# for configure options
+./something_similar.sh
+</pre>
+</td>
+</tr>
+
+
+<tr>
+<td colspan="2">
+<b>- Alternative: Build and install the entire library step by step:</b>
+</td>
+</tr>
+<tr>
+<td>
+<pre>
+# Go to home directory
+cd $oomph_home_dir<br>
+# Run manual config/build/install 
+# sequence
+./configure hierher check this
+make 
+make install
+</pre>
+</td>
+<td>
+<pre>
+# Go to home directory
+cd $oomph_home_dir<br>
+# Run manual config/build/install
+# sequence
+cmake -G Ninja -B build
+cmake --build build
+cmake --install build
+</pre>
+</td>
+</tr>
+
+<tr>
+<td colspan="2">
+<b>- Run all self-tests:</b>
+</td>
+</tr>
+<tr>
+<td>
+<pre>
+# Go to home directory
+cd $oomph_home_dir<br>
+# Run the tests
+make check
+</pre>
+</td>
+<td>
+<pre>
+# Go to demo_drivers directory hierher Puneet: does this also do the tests in self_tests?
+cd $oomph_home_dir/demo_drivers<br>
+# Configure
+cmake -G Ninja -B build<br>
+# Run the tests
+cd build
+ctest
+</pre>
+</td>
+</tr>
+
+<tr>
+<td colspan="2">
+<b>- Run all self-tests in a given demo driver directory:</b>
+</td>
+</tr>
+<tr>
+<td>
+<pre>
+# Go to demo driver directory
+cd $oomph_home_dir/demo_driver/one_d_poisson<br>
+# Run the test
+make check
+</pre>
+</td>
+<td>
+<pre>
+# Go to demo driver directory
+cd $oomph_home_dir/demo_driver/one_d_poisson<br>
+# Configure
+cmake -G Ninja -B build<br>
+# Run the test
+cd build
+ctest
+</pre>
+</td>
+</tr>
+
+
+<tr>
+<td colspan="2">
+<b>- Debug demo driver code:</b>
+</td>
+</tr>
+<tr>
+<td>
+<pre>
+# Go to demo driver directory
+cd $oomph_home_dir/demo_driver/one_d_poisson<br>
+# Run the test
+make check<br>
+# Edit
+emacs one_d_poisson.cc<br>
+# Run again (make realises that 
+# code needs to be recompiled)
+make check<br>
+# Keep going until it works...
+# (of course you may do the
+# compilation/re-run in your 
+# editor/IDE; see below)
+</pre>
+</td>
+<td>
+<pre>
+# Go to demo driver directory
+cd $oomph_home_dir/demo_driver/one_d_poisson<br>
+# Configure (note that switching on the
+# debug build only helps if the library
+# itself was compiled with that option too)
+cmake -G Ninja -B build -DCMAKE_BUILD_TYPE=Debug<br>
```

Comment:
Isn't "Release" the default?

Reply:
For the build of the main library... We don't set defaults in the separate demo driver projects (this would be some extra boilerplate that we would need to add -- and probably should now that you mention it...)

Reply:
**TODO:** Add warning/check to `oomphlibConfig.cmake` to make sure the build type is the same as the build of the main oomph-lib project.

Reply:
Done.